sonic-xcvrd/xcvrd/pm_mgr.py

```diff
@@ -0,0 +1,497 @@
+#!/usr/bin/env python3
+
+"""
+    Performance Monitoring statistics manager 
+"""
+
+try:
+    import ast
+    import copy
+    import sys
+    import threading
+    import time
+    import traceback
+    import sched
+
+    from swsscommon import swsscommon
+    from .xcvrd_utilities import sfp_status_helper
+    from .xcvrd_utilities.xcvr_table_helper import XcvrTableHelper
+    from .xcvrd_utilities import port_mapping
+except ImportError as e:
+    raise ImportError(str(e) + " - required module not found")
+
+#
+# Constants ====================================================================
+#
+
+PM_INFO_UPDATE_PERIOD_SECS = 60
+
+MAX_NUM_60SEC_WINDOW = 15
+MAX_NUM_15MIN_WINDOW = 12
+MAX_NUM_24HRS_WINDOW = 2
+
+WINDOW_60SEC_START_NUM = 1
+WINDOW_15MIN_START_NUM = 16
+WINDOW_24HRS_START_NUM = 28
+
+WINDOW_60SEC_END_NUM = 15
+WINDOW_15MIN_END_NUM = 27
+WINDOW_24HRS_END_NUM = 29
+
+TIME_60SEC_IN_SECS = 60 
+TIME_15MIN_IN_SECS = 900
+TIME_24HRS_IN_SECS = 86400 
+
+class PmUpdateTask(threading.Thread):
+
+    # Subscribe to below table in Redis DB
+    PORT_TBL_MAP = [
+        {'STATE_DB': 'TRANSCEIVER_INFO', 'FILTER': ['media_interface_code']},
+    ]
+
+    def __init__(self, namespaces, port_mapping_data, main_thread_stop_event, platform_chassis, helper_logger, process_start_option, pm_interval):
+        threading.Thread.__init__(self)
+        self.name = "PmUpdateTask"
+        self.exc = None
+        self.task_stopping_event = threading.Event()
+        self.main_thread_stop_event = main_thread_stop_event
+        self.port_mapping_data = copy.deepcopy(port_mapping_data)
```

Comment:
Since port_mapping is copied at the init time of xcvrd, what about the case of DPB, where port_mapping can get changed at runtime?
Since port_mapping's purpose here is to get the port_index/asic_index later in runtime of PM thread, another way to get port_index/asic_index can be from on_port_update_event if registering to CONFIG_DB PORT_TABLE.